rust/connlib/snownet/src/node.rs

```diff
@@ -1178,13 +1190,67 @@ enum PeerSocket {
     Direct {
         source: SocketAddr,
         dest: SocketAddr,
+        dest_kind: CandidateKind,
     },
     Relay {
         relay: SocketAddr,
         dest: SocketAddr,
+        dest_kind: CandidateKind,
     },
 }
 
+impl PartialOrd for PeerSocket {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        match (self, other) {
+            (PeerSocket::Direct { .. }, PeerSocket::Relay { .. }) => {
+                Some(std::cmp::Ordering::Greater)
+            }
+            (PeerSocket::Relay { .. }, PeerSocket::Direct { .. }) => {
+                Some(std::cmp::Ordering::Greater)
+            }
+            (
+                PeerSocket::Direct {
+                    dest_kind: left, ..
+                },
+                PeerSocket::Direct {
+                    dest_kind: right, ..
+                },
+            )
+            | (
+                PeerSocket::Relay {
+                    dest_kind: left, ..
+                },
+                PeerSocket::Relay {
+                    dest_kind: right, ..
+                },
+            ) => Some(cmp_candidate_kind(left, right)),
+        }
+    }
+}
+
+fn cmp_candidate_kind(left: &CandidateKind, right: &CandidateKind) -> std::cmp::Ordering {
+    use std::cmp::Ordering::*;
+
+    match (left, right) {
+        (CandidateKind::Host, CandidateKind::PeerReflexive) => Greater,
+        (CandidateKind::Host, CandidateKind::ServerReflexive) => Greater,
+        (CandidateKind::Host, CandidateKind::Relayed) => Greater,
+        (CandidateKind::PeerReflexive, CandidateKind::Relayed) => Greater,
+        (CandidateKind::ServerReflexive, CandidateKind::Relayed) => Greater,
+        (CandidateKind::PeerReflexive, CandidateKind::Host) => Less,
+        (CandidateKind::ServerReflexive, CandidateKind::Host) => Less,
+        (CandidateKind::Relayed, CandidateKind::Host) => Less,
+        (CandidateKind::Relayed, CandidateKind::PeerReflexive) => Less,
+        (CandidateKind::Relayed, CandidateKind::ServerReflexive) => Less,
+        (CandidateKind::Host, CandidateKind::Host)
+        | (CandidateKind::Relayed, CandidateKind::Relayed)
+        | (CandidateKind::PeerReflexive, CandidateKind::PeerReflexive)
+        | (CandidateKind::ServerReflexive, CandidateKind::ServerReflexive)
+        | (CandidateKind::ServerReflexive, CandidateKind::PeerReflexive)
+        | (CandidateKind::PeerReflexive, CandidateKind::ServerReflexive) => Equal,
```

Comment:
Maybe we should prefer peer reflexive, as the [ICE RFC describes](https://datatracker.ietf.org/doc/html/rfc8445#appendix-B.7)